worker-core/src/main/java/com/github/workerframework/core/WorkerApplication.java

```diff
@@ -143,25 +145,27 @@ public void start() {
 
             @Override
             public void stop() {
-                LOG.info("Worker stop requested, allowing in-progress tasks to complete.");
+                LOG.info("Worker stop requested.");
+                
                 workerQueue.shutdownIncoming();
-                while (!wtp.isIdle()) {
+
+                final long startTime = System.currentTimeMillis();
+                
+                while(wtp.getBacklogSize() > 0 && System.currentTimeMillis() - startTime < SHUTDOWN_DURATION) {
                     try {
-                        //The grace period will expire and the process killed so no need for time limit here
-                        LOG.trace("Awaiting the Worker Thread Pool to become idle, {} tasks in the backlog.", 
-                                wtp.getBacklogSize());
-                        Thread.sleep(1000);
+                        LOG.info("Allowing {} backlog tasks to complete, {} currently active.", wtp.getBacklogSize(), wtp.getActiveCount());
+                        Thread.sleep(SHUTDOWN_LOG_INTERVAL); // 15 seconds
```

Comment:
Won't this cause normal shutdowns to take much longer?  Why do we want to move this from 1 second to 15 seconds?

Reply:
We won't enter the while loop if there's no backlog and we don't want to log too much so normal shutdown will be ok?

Reply:
A backlog being present will be a regular event.  To avoid logging too much could we just put the log statement outside the loop, or downgrade it to debug, or add some extra logic so that it is not logged in every iteration.